spidermon/contrib/scrapy/monitors.py

```diff
@@ -87,24 +88,28 @@ def test_should_finish_with_expected_reason(self):
 class UnwantedHTTPCodesMonitor(BaseScrapyMonitor):
     """Check for maximum number of unwanted HTTP codes.
 
-    You can configure a ``dict`` of unwanted HTTP codes with
-    ``SPIDERMON_UNWANTED_HTTP_CODES`` the default value is::
+    You can configure a ``list`` of unwanted HTTP codes with
+    ``SPIDERMON_UNWANTED_HTTP_CODES`` and ``SPIDERMON_UNWANTED_HTTP_CODES_THRESHOLD``
+     the default values are::
 
-        DEFAULT_ERROR_CODES = {
-            code: 10
-            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]}
+        DEFAULT_ERROR_CODES = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        DEFAULT_ERROR_THRESHOLD = 10
     """
 
-    DEFAULT_ERROR_CODES = {
-        code: 10 for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
-    }
+    DEFAULT_ERROR_CODES = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+    DEFAULT_ERROR_THRESHOLD = 10
 
     @monitors.name("Should not hit the limit of unwanted http status")
     def test_check_unwanted_http_codes(self):
-        error_codes = self.crawler.settings.getdict(
+        error_codes = self.crawler.settings.getlist(
```

Comment:
This breaks backwards compatibility.
Maybe:
* try to read a dict (keeps it backwards compatible and enables us to configure different counts for each error)
* If it is a dict, just keep it like that
* If it is a list, create a dict using the codes in the list and `SPIDERMON_UNWANTED_HTTP_CODES_THRESHOLD`

Reply:
Oh, supporting both lists and dicts, great idea.

As part of https://github.com/scrapy/scrapy/pull/3696 Scrapy’s `Setting` class is getting a `getdictorlist` method, but since you probably do not want to wait for that to be merged or have Spidermon require some future Scrapy version, you might need to hardcode something in the lines of `getdictorlist` here.